scylla-cdc-replicator/src/main.rs

```diff
@@ -1,29 +1,68 @@
 mod replication_tests;
+mod replicator;
 pub mod replicator_consumer;
 
-use replicator_consumer::ReplicatorConsumerFactory;
-use scylla::SessionBuilder;
-use std::error::Error;
-use std::sync::Arc;
 use std::time::Duration;
 
+use clap::Parser;
+use futures_util::StreamExt;
+
+use crate::replicator::Replicator;
+
+#[derive(Parser)]
+struct Args {
+    /// Keyspace name
+    #[clap(short, long)]
+    keyspace: String,
+
+    /// Table names provided as a comma delimited string
+    #[clap(short, long)]
+    table: String,
+
+    /// Address of a node in source cluster
+    #[clap(short, long)]
+    source: String,
+
+    /// Address of a node in destination cluster
+    #[clap(short, long)]
+    destination: String,
+
+    /// Window size in seconds
+    #[clap(long, default_value_t = 60.)]
+    window_size: f64,
+
+    /// Safety interval in seconds
+    #[clap(long, default_value_t = 30.)]
+    safety_interval: f64,
+
+    /// Sleep interval in seconds
+    #[clap(long, default_value_t = 10.)]
+    sleep_interval: f64,
+}
+
 #[tokio::main]
-async fn main() -> Result<(), Box<dyn Error>> {
-    println!("Hello");
-    tokio::time::sleep(Duration::from_secs(1)).await;
-
-    // Create a factory to remove the warning about unused new() function.
-    let _factory = ReplicatorConsumerFactory::new(
-        Arc::new(
-            SessionBuilder::new()
-                .known_node("127.0.0.1:9042")
-                .build()
-                .await?,
-        ),
-        "ks".to_string(),
-        "t".to_string(),
-    );
-
-    println!("Replicator");
+async fn main() -> anyhow::Result<()> {
+    let args = Args::parse();
+    let tables: Vec<_> = args.table.split(',').map(|s| s.to_string()).collect();
+
+    let (mut replicator, mut handles) = Replicator::new(
+        args.source,
+        args.keyspace.clone(),
+        tables.clone(),
+        args.destination,
+        args.keyspace,
+        tables,
+        Duration::from_secs_f64(args.window_size),
+        Duration::from_secs_f64(args.safety_interval),
+        Duration::from_secs_f64(args.sleep_interval),
+    )
+    .await?;
+
+    tokio::signal::ctrl_c().await.unwrap();
+
+    println!("Shutting down...");
+
+    replicator.stop();
+    while handles.next().await.is_some() {}
```

Comment:
I think there is a problem here... If one of the replicators stops due to an error, the whole application won't notice unless the user presses Ctrl+C. The application should react immediately if this happens, i.e. stop other replicators and - ideally - wait until remaining replicators complete.

Moreover, you are not reporting the errors in any way, i.e. no error is returned from main and nothing is printed. Unless there is some logging in the replicator (which I'm not aware of) you won't even know why it stopped and if it was due to a failure or not.